package NYT.NCypressServer.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

import "yt_proto/yt/core/ytree/proto/attributes.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqCreatePortalExit
{
    required NYT.NProto.TGuid entrance_node_id = 1;
    required NYT.NProto.TGuid account_id  = 2;
    required string path = 3;
    required NYT.NYTree.NProto.TAttributeDictionary explicit_node_attributes = 4;
    required NYT.NYTree.NProto.TAttributeDictionary inherited_node_attributes = 5;
    required bytes acl = 6;
    required NYT.NProto.TGuid parent_id = 7;
    optional string key = 8;
    required string annotation = 9;
}

////////////////////////////////////////////////////////////////////////////////

message TReqRemovePortalEntrance
{
    required NYT.NProto.TGuid entrance_node_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqRemovePortalExit
{
    required NYT.NProto.TGuid exit_node_id = 1;
}

////////////////////////////////////////////////////////////////////////////////
